lib/registration.pm

```diff
@@ -274,15 +274,20 @@ sub fill_in_registration_data {
     # Process modules on sle 15
     if (is_sle '15+') {
         my $modules_needle = "modules-preselected-" . get_required_var('SLE_PRODUCT');
-        if (check_var('BETA', '1')) {
-            assert_screen('scc-beta-filter-checkbox');
-            send_key('alt-i');
-        }
-        elsif (!check_screen($modules_needle, 0)) {
-            record_info('bsc#1094457', 'bsc#1094457 : SLE 15 modules are still in BETA while product enter GMC phase');
-            assert_screen('scc-beta-filter-checkbox');
-            send_key('alt-i');
+        # Check needle 'scc-beta-filter-checkbox', if yes means product still in BETA phase, then continue assert process;
+        # if not means product already out of BETA phase, then do not need to assert the 'scc-beta-filter-checkbox' any more.
+        if (check_screen('scc-beta-filter-checkbox')) {
```

Comment:
I don't think that I like this code, we should be clear on when to expect this and when not. So I liked previous solution more except of missing small part. If we make logic based on the screen, I would propose to use multl-tag assert, or at least do some assertions before this `check_screen`, as with 0 timeout we might not catch it =(

Reply:
Rodion, I updated the code to use multi-tag to assert.
Added one needle to deal condition that no beta filter checkbox exist, please merge it together.
https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/1005
verify log:
http://openqa-apac1.suse.de/tests/1926#